test_bfs.cu:76:108: error: no matching function for call

On a system with no admin access, I faced this compilation error and I don't know if it is with cuda, gunrock, boost or gcc.

I have compiled gunrock-1.0 with cuda-10 before, but this time, the specs are

gcc 4.8.6
boost 1.65.1
cuda 10.0
gunrock 1.0

The error is 
```
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh:504:102:   required from ‘static cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160u>::Act(gunrock::util::Parameters&, OpT) [with OpT = main_struct; VertexT = long unsigned int; unsigned int FLAG = 197290u; cudaError_t = cudaError]’
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh:624:91:   required from ‘static cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10u>::Act(gunrock::util::Parameters&, OpT) [with OpT = main_struct; unsigned int FLAG = 197290u; cudaError_t = cudaError]’
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh:641:80:   required from ‘cudaError_t gunrock::app::Switch_Types(gunrock::util::Parameters&, OpT) [with unsigned int FLAG = 197290u; OpT = main_struct; cudaError_t = cudaError]’
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu:120:62:   required from here
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu:76:108: error: no matching function for call to ‘CPU_Reference(GraphT&, long unsigned int&, bool&, bool, LabelT*&, NULL, long unsigned int&)’
         float elapsed = app::bfs::CPU_Reference(graph, src, quiet, false,
                                                                                                            ^
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu:76:108: note: candidate is:
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_test.cuh:60:1: note: template<class GraphT, class LabelT> double gunrock::app::bfs::CPU_Reference(const GraphT&, typename GraphT::VertexT, bool, bool, LabelT*, typename GraphT::VertexT*, typename GraphT::VertexT&)
 double CPU_Reference(const GraphT &graph, typename GraphT::VertexT src,
 ^
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_test.cuh:60:1: note:   template argument deduction/substitution failed:
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu:76:108: note:   mismatched types ‘typename GraphT::VertexT*’ and ‘long int’
         float elapsed = app::bfs::CPU_Reference(graph, src, quiet, false,
                                                                                                            ^
```
Can someone help on who is causing this error?

It is confusing.... I have to use `GraphT::VertexT *ref_preds = (GraphT::VertexT*)NULL`
There are two `Validate_Results` in bfs_app.cu and I have casted in a way that I think it is correct.

```
    if (validation == "each") {
      GUARD_CU(problem.Extract(h_labels, h_preds));
      SizeT num_errors = app::bfs::Validate_Results(
          parameters, graph, src, h_labels, h_preds,
          ref_labels == (LabelT*)NULL ? (LabelT*)NULL : ref_labels[run_num % num_srcs], (VertexT*)NULL,
          false);
    }
  }

  cpu_timer.Start();
  // Copy out results
  GUARD_CU(problem.Extract(h_labels, h_preds));
  if (validation == "last") {
    SizeT num_errors = app::bfs::Validate_Results(
        parameters, graph, src, h_labels, h_preds,
        ref_labels == (LabelT*)NULL ? (LabelT*)NULL : ref_labels[(num_runs - 1) % num_srcs]);
  }
```
I also modified the source of `bfs_test.cu` as below:

```
typename GraphT::SizeT Validate_Results(
    util::Parameters &parameters, GraphT &graph, typename GraphT::VertexT src,
    LabelT *h_labels, typename GraphT::VertexT *h_preds,
    LabelT *ref_labels = (LabelT*)NULL, typename GraphT::VertexT *ref_preds = (GraphT::VertexT*)NULL,
    bool verbose = true) {
```
Also, in `Validate_Results`, I modified:

```
if (ref_labels != (LabelT*)NULL) {
...
} else if (ref_labels == (LabelT*)NULL) {
...
if (ref_labels != (LabelT*)NULL) {
...
if (ref_preds != (VertexT*)NULL) {
```

Here is the error I get

```
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(125): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, VertexT, uint32_t *, VertexT *, <error-type>)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"

```
The full output is:
```
[ 12%] Building NVCC (Device) object examples/bfs/CMakeFiles/bfs.dir/bfs_generated_test_bfs.cu.o
/storage/users/mnaderan/cactus/gu/gunrock-1.0/externals/rapidjson/include/rapidjson/writer.h(550): warning: statement is unreachable

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_test.cuh(135): error: expected an expression

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_test.cuh(135): error: unexpected end of default argument expression

/storage/users/mnaderan/cactus/gu/gunrock-1.0/externals/rapidjson/include/rapidjson/writer.h(352): warning: statement is unreachable
          detected during:
            instantiation of "__nv_bool rapidjson::Writer<OutputStream, SourceEncoding, TargetEncoding, StackAllocator, writeFlags>::WriteDouble(double) [with OutputStream=rapidjson::FileWriteStream, SourceEncoding=rapidjson::UTF8<char>, TargetEncoding=rapidjson::UTF8<char>, StackAllocator=rapidjson::CrtAllocator, writeFlags=0U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/externals/rapidjson/include/rapidjson/prettywriter.h(101): here
            instantiation of "__nv_bool rapidjson::PrettyWriter<OutputStream, SourceEncoding, TargetEncoding, StackAllocator, writeFlags>::Double(double) [with OutputStream=rapidjson::FileWriteStream, SourceEncoding=rapidjson::UTF8<char>, TargetEncoding=rapidjson::UTF8<char>, StackAllocator=rapidjson::CrtAllocator, writeFlags=0U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/util/info_rapidjson.cuh(301): here
            instantiation of "void gunrock::util::Info::SetDouble(DummyT, const float &) [with DummyT=std::string]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/util/info_rapidjson.cuh(335): here
            instantiation of "void gunrock::util::Info::SetVal(std::string, const T &, __nv_bool) [with T=float]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/util/info_rapidjson.cuh(212): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graph/coo.cuh(397): warning: integer conversion resulted in a change of sign
          detected during:
            instantiation of "cudaError_t gunrock::graph::Coo<_VertexT, _SizeT, _ValueT, _FLAG, cudaHostRegisterFlag, VALID>::EdgeDouble(__nv_bool, __nv_bool) [with _VertexT=uint64_t, _SizeT=uint64_t, _ValueT=uint32_t, _FLAG=1792U, cudaHostRegisterFlag=0U, VALID=true]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(594): here
            instantiation of "cudaError_t gunrock::graphio::market::Read(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::graph::Coo<uint64_t, uint64_t, uint32_t, 1792U, 0U, true>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(642): here
            instantiation of "cudaError_t gunrock::graphio::market::CooSwitch<GraphT, false>::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(653): here
            instantiation of "cudaError_t gunrock::graphio::market::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/graphio.cuh(201): here
            instantiation of "cudaError_t gunrock::graphio::LoadGraph(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(54): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(116): error: operand types are incompatible ("uint64_t **" and "uint64_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(114): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>, VertexT, uint64_t *, VertexT *, <error-type>, VertexT *, __nv_bool)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(127): error: operand types are incompatible ("uint64_t **" and "uint64_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(125): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>, VertexT, uint64_t *, VertexT *, <error-type>)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graph/coo.cuh(397): warning: integer conversion resulted in a change of sign
          detected during:
            instantiation of "cudaError_t gunrock::graph::Coo<_VertexT, _SizeT, _ValueT, _FLAG, cudaHostRegisterFlag, VALID>::EdgeDouble(__nv_bool, __nv_bool) [with _VertexT=uint64_t, _SizeT=uint32_t, _ValueT=uint32_t, _FLAG=1792U, cudaHostRegisterFlag=0U, VALID=true]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(594): here
            instantiation of "cudaError_t gunrock::graphio::market::Read(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::graph::Coo<uint64_t, uint32_t, uint32_t, 1792U, 0U, true>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(642): here
            instantiation of "cudaError_t gunrock::graphio::market::CooSwitch<GraphT, false>::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(653): here
            instantiation of "cudaError_t gunrock::graphio::market::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/graphio.cuh(201): here
            instantiation of "cudaError_t gunrock::graphio::LoadGraph(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(54): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(116): error: operand types are incompatible ("uint64_t **" and "uint64_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(114): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>, VertexT, uint64_t *, VertexT *, <error-type>, VertexT *, __nv_bool)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(127): error: operand types are incompatible ("uint64_t **" and "uint64_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(125): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>, VertexT, uint64_t *, VertexT *, <error-type>)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint64_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint64_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(624): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graph/coo.cuh(397): warning: integer conversion resulted in a change of sign
          detected during:
            instantiation of "cudaError_t gunrock::graph::Coo<_VertexT, _SizeT, _ValueT, _FLAG, cudaHostRegisterFlag, VALID>::EdgeDouble(__nv_bool, __nv_bool) [with _VertexT=uint32_t, _SizeT=uint64_t, _ValueT=uint32_t, _FLAG=1792U, cudaHostRegisterFlag=0U, VALID=true]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(594): here
            instantiation of "cudaError_t gunrock::graphio::market::Read(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::graph::Coo<uint32_t, uint64_t, uint32_t, 1792U, 0U, true>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(642): here
            instantiation of "cudaError_t gunrock::graphio::market::CooSwitch<GraphT, false>::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(653): here
            instantiation of "cudaError_t gunrock::graphio::market::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/graphio.cuh(201): here
            instantiation of "cudaError_t gunrock::graphio::LoadGraph(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(54): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(116): error: operand types are incompatible ("uint32_t **" and "uint32_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(114): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>, VertexT, uint32_t *, VertexT *, <error-type>, VertexT *, __nv_bool)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(127): error: operand types are incompatible ("uint32_t **" and "uint32_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(125): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>, VertexT, uint32_t *, VertexT *, <error-type>)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint64_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint64_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(504): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graph/coo.cuh(397): warning: integer conversion resulted in a change of sign
          detected during:
            instantiation of "cudaError_t gunrock::graph::Coo<_VertexT, _SizeT, _ValueT, _FLAG, cudaHostRegisterFlag, VALID>::EdgeDouble(__nv_bool, __nv_bool) [with _VertexT=uint32_t, _SizeT=uint32_t, _ValueT=uint32_t, _FLAG=1792U, cudaHostRegisterFlag=0U, VALID=true]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(594): here
            instantiation of "cudaError_t gunrock::graphio::market::Read(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::graph::Coo<uint32_t, uint32_t, uint32_t, 1792U, 0U, true>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(642): here
            instantiation of "cudaError_t gunrock::graphio::market::CooSwitch<GraphT, false>::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/market.cuh(653): here
            instantiation of "cudaError_t gunrock::graphio::market::Load(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/graphio/graphio.cuh(201): here
            instantiation of "cudaError_t gunrock::graphio::LoadGraph(gunrock::util::Parameters &, GraphT &, std::string) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(54): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(116): error: operand types are incompatible ("uint32_t **" and "uint32_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(114): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, VertexT, uint32_t *, VertexT *, <error-type>, VertexT *, __nv_bool)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(127): error: operand types are incompatible ("uint32_t **" and "uint32_t *")
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/bfs/bfs_app.cu(125): error: no instance of function template "gunrock::app::bfs::Validate_Results" matches the argument list
            argument types are: (gunrock::util::Parameters, gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, VertexT, uint32_t *, VertexT *, <error-type>)
          detected during:
            instantiation of "cudaError_t gunrock::app::bfs::RunTests(gunrock::util::Parameters &, GraphT &, LabelT **, gunrock::util::Location) [with GraphT=gunrock::app::TestGraph<uint32_t, uint32_t, uint32_t, 768U, 0U>, LabelT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(89): here
            instantiation of "cudaError_t main_struct::operator()(gunrock::util::Parameters &, VertexT, SizeT, ValueT) [with VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(210): here
            instantiation of "cudaError_t gunrock::app::Switch_Direction<OpT,VertexT,SizeT,ValueT,FLAG>(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, ValueT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(266): here
            instantiation of "cudaError_t gunrock::app::Switch_ValueT<OpT, VertexT, SizeT, FLAG, 512U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, SizeT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(508): here
            instantiation of "cudaError_t gunrock::app::Switch_SizeT<OpT, VertexT, FLAG, 160U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, VertexT=uint32_t, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(628): here
            instantiation of "cudaError_t gunrock::app::Switch_VertexT<OpT, FLAG, 10U>::Act(gunrock::util::Parameters &, OpT) [with OpT=main_struct, FLAG=197290U]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/gunrock/app/test_base.cuh(641): here
            instantiation of "cudaError_t gunrock::app::Switch_Types<FLAG,OpT>(gunrock::util::Parameters &, OpT) [with FLAG=197290U, OpT=main_struct]"
/storage/users/mnaderan/cactus/gu/gunrock-1.0/examples/bfs/test_bfs.cu(120): here

18 errors detected in the compilation of "/tmp/tmpxft_00007353_00000000-6_test_bfs.cpp1.ii".
CMake Error at bfs_generated_test_bfs.cu.o.cmake:279 (message):
  Error generating file
  /storage/users/mnaderan/cactus/gu/gunrock-1.0/build/examples/bfs/CMakeFiles/bfs.dir//./bfs_generated_test_bfs.cu.o

```

--- a/examples/bfs/test_bfs.cu
+++ b/examples/bfs/test_bfs.cu
@@ -74,7 +74,7 @@ struct main_struct {
         VertexT num_iters = 0;
         util::PrintMsg("__________________________", !quiet);
         float elapsed = app::bfs::CPU_Reference(graph, src, quiet, false,
-                                                ref_labels[i], NULL, num_iters);
+                                                ref_labels[i], (VertexT*)NULL, num_iters);
         util::PrintMsg("--------------------------\nRun " + std::to_string(i) +
                            " elapsed: " + std::to_string(elapsed) +
                            " ms, src = " + std::to_string(src) +

--- a/examples/sssp/test_sssp.cu
+++ b/examples/sssp/test_sssp.cu
@@ -78,7 +78,7 @@ struct main_struct {
         VertexT src = srcs[i];
         util::PrintMsg("__________________________", !quiet);
         float elapsed = app::sssp::CPU_Reference(graph.csr(), ref_distances[i],
-                                                 NULL, src, quiet, false);
+                                                 (VertexT*)NULL, src, quiet, false);
         util::PrintMsg("--------------------------\nRun " + std::to_string(i) +
                            " elapsed: " + std::to_string(elapsed) +
                            " ms, src = " + std::to_string(src),

--- a/examples/vn/test_vn.cu
+++ b/examples/vn/test_vn.cu
@@ -133,7 +133,7 @@ struct main_struct {
 
         util::PrintMsg("__________________________", !quiet);
         float elapsed =
-            app::vn::CPU_Reference(graph.csr(), ref_distances[run_num], NULL,
+            app::vn::CPU_Reference(graph.csr(), ref_distances[run_num], (VertexT*)NULL,
                                    srcs, srcs_per_run, quiet, false);
 
         std::string src_msg = "";

--- a/gunrock/app/bfs/bfs_app.cu
+++ b/gunrock/app/bfs/bfs_app.cu
@@ -113,7 +113,7 @@ cudaError_t RunTests(util::Parameters &parameters, GraphT &graph,
       GUARD_CU(problem.Extract(h_labels, h_preds));
       SizeT num_errors = app::bfs::Validate_Results(
           parameters, graph, src, h_labels, h_preds,
-          ref_labels == NULL ? NULL : ref_labels[run_num % num_srcs], NULL,
+          ref_labels == NULL ? NULL : ref_labels[run_num % num_srcs], (VertexT*)NULL,
           false);
     }
   }

--- a/gunrock/app/sage/sage_test.cuh
+++ b/gunrock/app/sage/sage_test.cuh
@@ -159,8 +159,10 @@ double CPU_Reference(util::Parameters &para, const GraphT &graph,
       // std::vector <SizeT> edges_source_child;
       // SizeT edges_source_child[num_children_per_source];
       VertexT children[num_children_per_source];
-      float children_temp[Wa2_dim0] = {0.0};  // agg(h_B1^1)
-      float source_temp[Wf2_dim0] = {0.0};    // h_B2^1
+      float children_temp[Wa2_dim0];  // agg(h_B1^1)
+      memset( children_temp, 0, Wa2_dim0*sizeof(float) );
+      float source_temp[Wf2_dim0];    // h_B2^1
+      memset( source_temp, 0, Wf2_dim0*sizeof(float) );
       // float source_result [256] = {0.0}; // h_B2_2, result
       auto source_result =
           source_embedding + ((uint64_t)source) * result_column;
@@ -199,7 +201,8 @@ double CPU_Reference(util::Parameters &para, const GraphT &graph,
         // SizeT pos = edges_source_child[i];
         // VertexT child = graph.GetEdgeDest(pos);
         VertexT child = children[i];
-        float sums[feature_column] = {0.0};
+        float sums[feature_column];
+        memset( sums, 0, feature_column*sizeof(float) );
 
         // sample leaf node for each child
         for (int j = 0; j < num_leafs_per_child; j++) {
@@ -225,7 +228,8 @@ double CPU_Reference(util::Parameters &para, const GraphT &graph,
           sums[m] = sums[m] / num_leafs_per_child;
         }  // get mean  agg of leaf features.
         // get ebedding vector for child node (h_{B1}^{1}) alg2 line 12
-        float child_temp[Wa2_dim0] = {0.0};
+        float child_temp[Wa2_dim0];
+        memset( child_temp, 0, Wa2_dim0*sizeof(float) );
         for (int idx_0 = 0; idx_0 < Wf1_dim1; idx_0++) {
           for (int idx_1 = 0; idx_1 < feature_column; idx_1++)
             child_temp[idx_0] += features[child][idx_1] * W_f_1[idx_1][idx_0];
@@ -256,7 +260,8 @@ double CPU_Reference(util::Parameters &para, const GraphT &graph,
       //////////////////////////////////////////////////////////////////////////////////////
       // get h_B2^1, k =1 ; this time, child is like leaf, and source is like
       // child
-      float sums_child_feat[feature_column] = {0.0};  // agg(h_B1^0)
+      float sums_child_feat[feature_column];  // agg(h_B1^0)
+      memset( sums_child_feat, 0, feature_column*sizeof(float) );
       for (int i = 0; i < num_children_per_source; i++) {
         // SizeT pos = edges_source_child[i];
         // VertexT child = graph.GetEdgeDest(pos);

--- a/gunrock/app/sssp/sssp_app.cu
+++ b/gunrock/app/sssp/sssp_app.cu
@@ -114,7 +114,7 @@ cudaError_t RunTests(util::Parameters &parameters, GraphT &graph,
       SizeT num_errors = app::sssp::Validate_Results(
           parameters, graph, src, h_distances, h_preds,
           ref_distances == NULL ? NULL : ref_distances[run_num % num_srcs],
-          NULL, false);
+          (VertexT*)NULL, false);
     }
   }
 

--- a/gunrock/app/vn/vn_app.cu
+++ b/gunrock/app/vn/vn_app.cu
@@ -146,7 +146,7 @@ cudaError_t RunTests(util::Parameters &parameters, GraphT &graph,
       GUARD_CU(problem.Extract(h_distances, h_preds));
       SizeT num_errors = app::vn::Validate_Results(
           parameters, graph, srcs, h_distances, h_preds,
-          ref_distances == NULL ? NULL : ref_distances[run_num], NULL, false);
+          ref_distances == NULL ? NULL : ref_distances[run_num], (VertexT*)NULL, false);
     }
   }
 

